beacon_node/eth2_libp2p/src/peer_manager/mod.rs

```diff
@@ -940,19 +947,31 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
         let connected_peer_count = self.network_globals.connected_peers();
         if connected_peer_count > self.target_peers {
             //remove excess peers with the worst scores, but keep subnet peers
-            for (peer_id, _) in self
+            //must also ensure that the outbound-only peer count does not go below the minimum threshold
+            let mut peers_removed = 0;
+            let mut n_outbound_removed = 0;
+            for (peer_id, info) in self
                 .network_globals
                 .peers
                 .read()
                 .worst_connected_peers()
                 .iter()
                 .filter(|(_, info)| !info.has_future_duty())
-                .take(connected_peer_count - self.target_peers)
-                //we only need to disconnect peers with healthy scores, since the others got already
-                //disconnected in update_peer_scores
-                .filter(|(_, info)| info.score_state() == ScoreState::Healthy)
             {
-                disconnecting_peers.push(**peer_id);
+                if peers_removed == connected_peer_count - self.target_peers {
+                    break;
+                }
+                if info.is_outbound_only() {
+                    if n_outbound_removed < outbound_only_peer_count - min_outbound_only_target {
```

Comment:
`outbound_only_peer_count` has no guarantee to be greater than `min_outbound_only_target`

Reply:
is this an issue tho? the conditional would evaluate to false, and we would not disconnect the outbound-only peer, which is the intended behavior

Reply:
It is an issue. For one logic should not depend on overflowing. This line with the right conditions panics on debug an evaluates to `true`pretty much always in release (0-1 in release is 18446744073709551615, the max value for usize)
I'll check your code again to give a better advice on how to accomplish what you want

Reply:
Ah good point! If I change the logic to ```min_outbound_only_target < n_outbound_removed + outbound_only_peer_count``` this would then handle the overflow issue while keeping the logic right?

Reply:
see my comment above to Age, think changing the conditional to ```min_outbound_only_target < outbound_only_peer_count - n_outbound_removed``` should preserve the logic while preventing the panic